sql/moz-fx-data-shared-prod/adjust_derived/adjust_derived_v1/query.py

```diff
@@ -0,0 +1,278 @@
+Keys are the app_name, and app_token in the form:
+{"app_name":"<specific app name from Adjust dash>". "app_token":"<unique Adjust app token>}"
+Look here to see the apps we are tracking: https://dash.adjust.com/#/.
+It is important to maintain this list in order for the script to work especially in the case of new apps
+being added to track in Adjust
+"""
+
+APP_TOKEN_LIST = os.environ.get("ADJUST_APP_TOKEN_LIST")
+
+API_TOKEN = os.environ.get("ADJUST_API_TOKEN")
+
+CSV_FIELDS = [
+    "date",
+    "app",
+    "app_token",
+    "network",
+    "network_token",
+    "campaign",
+    "campaign_token",
+    "adgroup",
+    "adgroup_token",
+    "creative",
+    "creative_token",
+    "country",
+    "os",
+    "device",
+    "clicks",
+    "installs",
+    "limit_ad_tracking_install_rate",
+    "click_conversion_rate",
+    "impression_conversion_rate",
+    "sessions",
+    "daus",
+    "waus",
+    "maus",
+]
+
+
+def read_json(filename: str) -> dict:
+    """Read JSON file."""
+    with open(filename, "r") as f:
+        data = json.loads(f.read())
+    return data
+
+
+def write_dict_to_csv(json_data, filename):
+    """Write a dictionary to a csv."""
+    with open(filename, "w") as out_file:
+        dict_writer = csv.DictWriter(out_file, CSV_FIELDS)
+        dict_writer.writeheader()
+        dict_writer.writerows(json_data)
+
+
+def download_adjust_kpi_data(date, api_token, app_token):
+    """Download data from Adjust - API token and APP tokens are called here."""
+    start_date = date
+    end_date = date
+
+    kpis = [
+        "clicks",
+        "installs",
+        "limit_ad_tracking_install_rate",
+        "click_conversion_rate",
+        "impression_conversion_rate",
+        "sessions",
+        "daus",
+        "waus",
+        "maus",
+    ]
+    groupings = [
+        "days",
+        "apps",
+        "networks",
+        "campaigns",
+        "adgroups",
+        "creatives",
+        "countries",
+        "os_names",
+        "device_types",
+    ]
+    # getting overview metrics for different kpis / Deliverables
+    url = f"https://api.adjust.com/kpis/v1/{app_token}"  # overview
+    url_params = f"start_date={start_date}&end_date={end_date}&kpis={','.join(kpis)}&grouping={','.join(groupings)}"
+    headers = {
+        "Authorization": f"Token token={api_token}",
+    }
+
+    response = requests.get(url, headers=headers, params=url_params)
+
+    return response
+
+
+def check_json(adjust_response_text):
+    """Script will return an empty dictionary for apps on days when there is no data. Check for that here."""
+    with tempfile.NamedTemporaryFile() as tmp_json:
+        with open(tmp_json.name, "w") as f_json:
+            f_json.write(adjust_response_text)
+            try:
+                query_export = read_json(f_json.name)
+            except (
+                ValueError
+            ):  # ex. json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
+                return None
```

Comment:
Should the application exit at this stop execution at this point? The error suggests like there might be a problem with the data?

Reply:
This was the solution to yesterday's issue where if there is no data, the script would get an empty dictionary. By returning None, there is an 'issue' of no data, but we don't always get data for each day.